rustls/src/msgs/handshake.rs

```diff
@@ -358,10 +358,10 @@ pub struct KeyShareEntry {
 }
 
 impl KeyShareEntry {
-    pub fn new(group: NamedGroup, payload: &[u8]) -> Self {
+    pub fn new(group: NamedGroup, payload: impl Into<Vec<u8>>) -> Self {
```

Comment:
I think we can ignore semver compat here because `KeyShareEntry` is only exposed via the `internal` namespace.